### PR TITLE
Use correct accessor for TEXT pointer

### PR DESCRIPTION
**Pull request by danielgustafsson**
CStringGetDatum operates on const char pointers, but restore_name is a TEXT pointer. Use the PointerGetDatum accessor instead. This fixes a compiler warning on passing incompatible pointer types.

While in there, fix copy/pasto in the file identification.

Note: will fix the commit message before pushing if accepted.

### CODE REVIEW

File: src/backend/access/transam/xlogfuncs_gp.c [hunk 1]
```diff
@@ -154,7 +154,7 @@ gp_create_restore_point(PG_FUNCTION_ARGS)
 		MemSet(nulls, false, sizeof(nulls));
 
 		values[0] = Int16GetDatum(seg_index);
-		values[1] = CStringGetDatum(restore_ptr);
+		values[1] = PointerGetDatum(restore_ptr);
```
Comment (hlinnaka) [inline src/backend/access/transam/xlogfuncs_gp.c]: PointerGetDatum() doesn't seem right here, either. `restore_ptr` is not a pointer, it's an XLogRecPtr, which means uint64.

Comment (hlinnaka) [reply src/backend/access/transam/xlogfuncs_gp.c]: It should be `LSNGetDatum(restore_ptr)'

Comment (danielgustafsson) [reply src/backend/access/transam/xlogfuncs_gp.c]: Of course.. I confused myself on the similarity of the variable names in the previous hunk where it is a TEXT pointer called `restore_name`. Fixing.

File: src/backend/access/transam/xlogfuncs_gp.c [hunk 2]
```diff
@@ -97,7 +97,7 @@ gp_create_restore_point(PG_FUNCTION_ARGS)
 						   DF_NEED_TWO_PHASE | DF_CANCEL_ON_ERROR,
 						   &context->cdb_pgresults);
 		context->qd_restorepoint_lsn = DirectFunctionCall1(pg_create_restore_point,
-														   CStringGetDatum(restore_name));
+														   PointerGetDatum(restore_name));
 		LWLockRelease(TwophaseCommitLock);
```
Comment (hlinnaka) [inline src/backend/access/transam/xlogfuncs_gp.c]: This isn't quite right either. `qd_restorepoint_lsn` is of type XLogRecPtr, but DirectFunctionCall1 returns a Datum. Should be:

```
context->qd_restorepoint_lsn = DatumGetLSN(DirectFunctionCall1(pg_create_restore_point,
														   CStringGetDatum(restore_name)));
```

Comment (danielgustafsson) [reply src/backend/access/transam/xlogfuncs_gp.c]: Good point, I was just looking at the warning on the input parameters and didn't consider that. Fixing too.

File: src/backend/access/transam/xlogfuncs_gp.c [hunk 3]
```diff
@@ -96,8 +96,8 @@ gp_create_restore_point(PG_FUNCTION_ARGS)
 		CdbDispatchCommand(restore_command,
 						   DF_NEED_TWO_PHASE | DF_CANCEL_ON_ERROR,
 						   &context->cdb_pgresults);
-		context->qd_restorepoint_lsn = DirectFunctionCall1(pg_create_restore_point,
-														   CStringGetDatum(restore_name));
+		context->qd_restorepoint_lsn = DatumGetLSN(DirectFunctionCall1(pg_create_restore_point,
+																	   PointerGetDatum(restore_name)));
```
Comment (soumyadeep2007) [inline src/backend/access/transam/xlogfuncs_gp.c]: This should be `CStringGetDatum` by convention as opposed to `PointerGetDatum` even though they are equivalent internally.

Comment (hlinnaka) [reply src/backend/access/transam/xlogfuncs_gp.c]: No, `restore_name` is of type `text *`, not `char *`.

**Comment by danielgustafsson** [comment]
Pushed to master, thanks for review!

**Pull request closed without merging**